pkg/volume/storageos/storageos_util.go

```diff
@@ -384,11 +389,22 @@ func getLoopDevice(path string, exec utilexec.Interface) (string, error) {
 func makeLoopDevice(path string, exec utilexec.Interface) (string, error) {
 	args := []string{"-f", "-P", "--show", path}
 	out, err := exec.Command(losetupPath, args...).CombinedOutput()
+	if err != nil {
+		if !losetupInvalidOpt(out) {
+			klog.V(2).Infof("Failed device create command for path %s: %v", path, err)
+			return "", err
+		}
```

Comment:
Would it make more sense to just print  stdout/stderr returned by CombinedOutput and not try to analyze what it means?

Reply:
Are you referring to printing `CombinedOutput` in `Infof` (which is a good idea) or checking the output? If it's the latter, we need it for checking if we have to lookup from `sysfs`.

Reply:
Just print whole `CombinedOutput` on error and do not parse it.

Reply:
@jsafrane - The reason for parsing the output for an invalid argument is to conditionally call `getLoopDeviceFromSysfs`. If command failed for any other reason (like `EBUSY`, or something similar), shouldn't we just return the failure?

BTW, I have updated the function (in 3f15e65) to print the output in the logs.

Reply:
> The reason for parsing the output for an invalid argument is to conditionally call getLoopDeviceFromSysfs. If command failed for any other reason (like EBUSY, or something similar), shouldn't we just return the failure?

I don't expect `losetup -j` to return `EBUSY`. As you wrote below, it probably does read the same sysfs as your `getLoopDeviceFromSysfs`.

Problem with parsing the error message of `losetup` is that it may change unexpectedly in the future and it depends on current locale. We do parse error messages elsewhere in Kubernetes, but it's not a good idea to spread it more.

Reply:
You're right, it would not return `EBUSY` for `-j`, for other options like `-f` it is a possibility. However, I do see that it becomes difficult to maintain of the the output is changed in the future.

I will remove `losetupInvalidOpt` and retry if it fails the first time. There's a good chance that if it failed for reasons other than an invalid option, it might fail again.